src/utils/utils.cpp

```diff
@@ -54,18 +55,51 @@ QDir mp::utils::base_dir(const QString& path)
     return info.absoluteDir();
 }
 
-bool mp::utils::valid_memory_value(const QString& mem_string)
+auto mp::utils::in_bytes(const std::string& mem_value) -> optional<long long>
 {
-    QRegExp matcher("\\d+((K|M|G)(B){0,1}){0,1}$");
+    static constexpr auto kilo = 1024LL;
 
-    return matcher.exactMatch(mem_string);
+    QRegExp matcher("^(\\d+)([KMG])?B?$", Qt::CaseInsensitive);
+
+    if (matcher.exactMatch(QString::fromStdString(mem_value)))
+    {
+        auto val =
+            std::stoll(matcher.cap(1).toStdString()); // value is in the second capture (1st one is the whole match)
```

Comment:
Qt has a QString::toLongLong

Reply:
Thank you, I did not know that. Changing accordingly

Reply:
I did this, so resolving